bot/exts/help_channels/_message.py

```diff
@@ -42,25 +42,28 @@
 """
 
 
-async def check_for_answer(message: discord.Message) -> None:
-    """Checks for whether new content in a help channel comes from non-claimants."""
+async def update_message_caches(message: discord.Message) -> None:
+    """Checks the source of new content in a help channel and updates the appropriate cache."""
     channel = message.channel
 
     # Confirm the channel is an in use help channel
     if is_in_category(channel, constants.Categories.help_in_use):
-        log.trace(f"Checking if #{channel} ({channel.id}) has been answered.")
-
-        # Check if there is an entry in unanswered
-        if await _caches.unanswered.contains(channel.id):
-            claimant_id = await _caches.claimants.get(channel.id)
-            if not claimant_id:
-                # The mapping for this channel doesn't exist, we can't do anything.
-                return
-
-            # Check the message did not come from the claimant
-            if claimant_id != message.author.id:
-                # Mark the channel as answered
-                await _caches.unanswered.set(channel.id, False)
+        log.trace(f"Checking if #{channel} ({channel.id}) has had a reply.")
+        # Must use a timezone-aware datetime to ensure a correct POSIX timestamp.
+        timestamp = datetime.now(timezone.utc).timestamp()
+
+        # Overwrite the claimant message time, if its from the claimant.
+        if message.author == await _caches.claimants.get(channel.id):
+            await _caches.claimant_last_message_times.set(channel.id, timestamp)
+            return
+
+        claimant_id = await _caches.claimants.get(channel.id)
```

Comment:
The claimant ID is being fetched twice. That can be avoided by saving the first value.

Reply:
it seems this would also check the author object against the id which even if discord.py handles that (not sure) is not as explicit as checking the id.
```suggestion
        claimant_id = await _caches.claimants.get(channel.id)

        if message.author.id == claimant_id:
            await _caches.claimant_last_message_times.set(channel.id, timestamp)
            return
```